### ISSUE TITLE
Rank Choice Ejection (with 4's)

### ISSUE THREAD

**Issue by Jayhui-q** [hanabi/hanabi.github.io#639]
Much like how a player can use color or rank on a 5 that is not on chop to get two different plays, I propose that this can be done in a similar way for 4s. 

### Rank Choice Ejection (with 4's)
- If Alice has a free choice between a 4 rank clue and a color clue (picking up exactly the same cards) when giving a 4 charm, then Alice's choice of a 4 rank clue is very odd.
- Indeed, had Alice chosen a color clue, Bob's charm would tell the clue recipient exactly what card they have. 
- Thus, if Alice chose rank instead of color, it is suboptimal, and must be calling for something else.
- We agree that Bob should eject instead.

For example, 
- In a 4-player no variant game, all 1s are on the stacks except for purple 1
- Cathy's hand holds b5 - p4 - g3 - g1
- Alice clues Cathy with 4.
- Bob sees that both a purple clue and a 4 clue would have resulted in a charm, since either clue would demand three blind-plays and Cathy does not have a playable or one-away-from-playable on finesse. Thus the 4 clue means something different.
- Instead, Bob ejects, and correctly plays blue 2.

tl;dr: give a 4 charm with rank when you could have used color -> ejection.
